cpp/daal/src/data_management/service_numeric_table.h

```diff
@@ -576,12 +591,29 @@ class GetRowsCSR
     {
         _status        = _data->getSparseBlock(iStartFrom, nRows, mode, _block);
         _toReleaseFlag = _status.ok();
+
+        if (_toOneBaseRowIndices)
+        {
+            if (_rowOffsets.size() < nRows + 1)
+            {
+                _rowOffsets.reset(nRows + 1);
+            }
+            const size_t * const rows = _block.getBlockRowIndicesPtr();
+            _rowOffsets[0]            = 1;
+            for (size_t i = 0; i < nRows; ++i)
+            {
+                const size_t nNonZeroValuesInRow = rows[i + 1] - rows[i];
+                _rowOffsets[i + 1]               = _rowOffsets[i] + nNonZeroValuesInRow;
+            }
+        }
     }
```

Comment:
We need convert _rowOffsets  only when mode != readOnly

Reply:
No, for example we create sub table for read accesses and we need aligned to one base indices.